Include/cpython/bytesobject.h

```diff
@@ -5,7 +5,7 @@
 typedef struct {
     PyObject_VAR_HEAD
     Py_hash_t ob_shash;
-    char ob_sval[1];
+    char ob_sval[];
```

Comment:
I suggest to:

* Leave the structure as it is
* Use a code search to check which C extension access directly PyBytesObject structure members (well, first check Cython and pybind11)
* Prepare affected C extensions
* Once the number of affected C extensions is low enough: make the structure opaque, enforce the usage of public C functions.